hyperspy/drawing/utils.py

```diff
@@ -333,6 +334,188 @@ def _set_spectrum_xlabel(spectrum, ax):
+                         "Number of signals was: " + repr(len(signals)) + ".")
+
+    # Determine appropriate number of images per row
+    n = len(signals)
+    rows = int(np.ceil(n / float(per_row)))
+    if n < per_row:
+            per_row = n
+
+    # Set overall figure size and define figure (if not pre-existing)
+    if fig is None:
+        k = max(plt.rcParams['figure.figsize']) / max(per_row, rows)
+        f = plt.figure(figsize=(tuple(k * i for i in (per_row, rows))))
+    else:
+        f = fig
+
+    # Initialize list to hold subplot axes
+    axes_list = []
+
+    # If using a single colorbar, find global min and max values of all the images
+    if single_colorbar:
+        gl_max, gl_min = max([signals[i].data.max() for i in range(len(signals))]), \
+                         min([signals[i].data.min() for i in range(len(signals))])
+
+    # Loop through each image, adding subplot for each one
+    for i in xrange(n):
+        ax = f.add_subplot(rows, per_row, i + 1)
+        axes_list.append(ax)
+        data = signals[i].data.flatten()
+        if no_nans:
+            data = np.nan_to_num(data)
+
+        axes_manager = sig.axes_manager
+        axes = axes_manager.signal_axes
+
+        if axes_manager.signal_dimension == 2:
+            extent = None
+            # get calibration from a passed axes_manager
+            shape = axes_manager._signal_shape_in_array
+
+            # Set dimensions of images
+            extent = (axes[0].low_value,
+                      axes[0].high_value,
+                      axes[1].high_value,
+                      axes[1].low_value)
+
+            # Plot image data, using vmin and vmax to set bounds, or allowing them
+            # to be set automatically if using individual colorbars
+            if single_colorbar:
+                im = ax.imshow(data.reshape(shape),
+                               cmap=cmap, extent=extent,
+                               interpolation='nearest',
+                               vmin=gl_min, vmax=gl_max)
+            else:
+                im = ax.imshow(data.reshape(shape),
+                               cmap=cmap, extent=extent,
+                               interpolation='nearest')
+
+            # Label the axes
+            plt.xlabel(axes[0].units)
+            plt.ylabel(axes[1].units)
+
+            if labels_on:
+                plt.title(label_list[i])
+
+            if not axes_on:
+                plt.axis('off')
+
+            # If using independent colorbars, add them
+            if plot_colorbar and not single_colorbar:
+                div = make_axes_locatable(ax)
+                cax = div.append_axes("right", size="5%", pad=0.05)
+                plt.colorbar(im, cax=cax)
+
+    # If using a single colorbar, add it
+    if plot_colorbar and single_colorbar:
+        f.subplots_adjust(right=0.8)
+        cbar_ax = f.add_axes([0.9, 0.1, 0.03, 0.8])
```

Comment:
This is not very flexible, and leads to the colorbar overlapping some of the images if the `per_row` value is high enough (see figure). There must be a better way to do this that makes it more flexible?

![figure_1](https://cloud.githubusercontent.com/assets/1278301/5967173/75b64dc0-a7d8-11e4-9a53-bbc6c1cd5a79.png)